apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/BigtableServiceApiClientPartial.cs

```diff
@@ -133,4 +187,383 @@ public partial class BigtableServiceApiClientImpl
                 request.IsIdempotent(),
                 "Non-idempotent MutateRows requests are not allowed. Specify a version with all SetCell mutations.");
     }
+
+    // TODO: Move changes into Google.Api.Gax.Grpc.ChannelPool if approved.
```

Comment:
Is this backward-compatible with the current API?

One thing we need to think about is making GAX depend on Grpc.Gcp. I need to look at more details of that package (and possibly discuss with other languages) about that. (It's not well-defined whether GAX is only intended for Google-hosted APIs, and I assume that "GCP" is really "Google-hosted" rather than "Part of Google Cloud Platform" here. (For example, it should be okay for a YouTube API with a gRPC endpoint, right?)

Reply:
Yes, I made no changes to the existing ChannelPool API.

I'm not sure about whether things can work with any Google-hosted API. @WeiranFang, can Grpc.Gcp be used with non-GCP endpoints?

Reply:
Hi @jskeet and @evildour , the Grpc.Gcp library is only for fixing GCP specific issues. We try to make the library as generic as possible, but the functionality is mainly designed for cloudpath (supported by CFE, a modified version of GFE). So the answer is Grpc.Gcp should be used only for GCP endpoints. Does that answer your question?

Reply:
Yes, thank you.